### PR TITLE
CODAP-1376: stabilize flaky case-table-keyboard Escape test

### PR DESCRIPTION
**Pull request by kswenson**
## Summary

Fixes the flaky Cypress test "Escape in SELECT mode blurs the focused cell"
(case-table-keyboard.spec.ts, added with CODAP-1365 / #2608). The flake was a
CI-only failure caused by a real edge-case gap in the production Escape handler,
so this includes a small app fix plus a test-precondition tweak.

## Root cause

The test failed only under code-coverage instrumentation on CI. Under that
slowdown, RDG momentarily reported the selected cell's `args.rowIdx` as a
negative sentinel (-2) right after the click while the cell was already focused.
`handleCellKeyDown` has an early `if (args.rowIdx < 0) return` guard *before* the
SELECT-mode Escape branch, so the blur never ran and focus stayed trapped on the
cell (`activeElement` = the grid cell, never `BODY`). Without coverage the rowIdx
was correct (>=0) and the blur ran — hence it passed locally and only flaked on CI.
(Confirmed via CI-only instrumentation: the Escape keydown reached the handler in
SELECT mode but bailed at the rowIdx guard; no focusout fired — it was not a
re-focus race.)

## Fix

- `collection-table.tsx`: handle SELECT-mode Escape **before** the `rowIdx < 0`
  guard so the blur runs regardless of the transient row index, guarded to only
  blur when focus is within the grid (so keys bubbling from portaled UI like the
  index-column menu aren't disturbed).
- `case-table-keyboard.spec.ts`: assert real DOM focus (`cy.focused()`) before
  pressing Escape instead of the `aria-selected` selection-model assertion.

## Testing

Validated on CI under the exact failing config (coverage on) and in a full
15-container regression attempt. Locally: lint, tsc, and the spec all green.

Fixes CODAP-1376

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: v3/src/components/case-table/collection-table.tsx [hunk 1]
```diff
@@ -281,6 +281,16 @@ export const CollectionTable = observer(function CollectionTable(props: IProps)
...
+    // Escape in SELECT mode blurs the focused cell (RDG's default only clears the copied-cell
+    // marker). Handle it BEFORE the `rowIdx < 0` guard below: under heavy load (e.g. code-coverage
+    // instrumentation on CI) RDG can momentarily report the selected position's rowIdx as negative
+    // right after a click while the cell is already focused. The guard would then skip the blur
+    // entirely, leaving focus trapped on the cell.
+    if (args.mode === "SELECT" && event.key === "Escape") {
+      event.preventGridDefault()
+      if (document.activeElement instanceof HTMLElement) document.activeElement.blur()
+      return
+    }
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/src/components/case-table/collection-table.tsx:299]: Good catch — the blur now only runs when `document.activeElement` is within `event.currentTarget` (the grid), so Escapes bubbling up from portaled UI like the index-column menu won't have their focus stolen. The CI flake scenario still blurs since the focused cell is inside the grid.

File: v3/src/components/case-table/collection-table.tsx [hunk 2]
```diff
@@ -281,6 +281,16 @@ export const CollectionTable = observer(function CollectionTable(props: IProps)
       event.preventGridDefault()
       return
     }
+    // Escape in SELECT mode blurs the focused cell (RDG's default only clears the copied-cell
+    // marker). Handle it BEFORE the `rowIdx < 0` guard below: under heavy load (e.g. code-coverage
+    // instrumentation on CI) RDG can momentarily report the selected position's rowIdx as negative
+    // right after a click while the cell is already focused. The guard would then skip the blur
+    // entirely, leaving focus trapped on the cell.
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/src/components/case-table/collection-table.tsx:288]: Good point — updated the PR description to reflect the production change: the real fix moves SELECT-mode Escape ahead of the `rowIdx<0` guard (rowIdx was transiently -2 under coverage instrumentation), with the `cy.focused()` test tweak as secondary hardening.

**Comment by kswenson** [comment]
Merging without review as it's just a tweak to fix a flaky cypress test.
